bouncer/src/repo/manipulator/modelconvertor/export/repo_model_export_src.cpp

```diff
@@ -164,8 +176,25 @@ std::unordered_map<std::string, std::vector<uint8_t>> SRCModelExport::getSRCFile
 
 		if (fdIt != fullDataBuffer.end())
 		{
+			auto fullDataArray = fdIt->second;
+
+			if (enableCompression) {
+				boost::iostreams::filtering_streambuf<boost::iostreams::input> out;
+				out.push(boost::iostreams::zlib_compressor());
+				out.push(boost::iostreams::array_source((const char*)fullDataArray.data(), fullDataArray.size()));
+
+				// The first 4 bytes define the length of the uncompressed data. ZLib, at the least, requires this be known ahead of time.
+				std::vector<uint8_t> compressed;
+				compressed.resize(4);
+				((uint32_t*)compressed.data())[0] = fullDataArray.size();
+				
+				compressed.insert(compressed.end(), std::istreambuf_iterator<char>(&out), std::istreambuf_iterator<char>());
```

Comment:
would it not make more sense to just insert the size number into the fullDataArray, at compressed.begin(), instead of copying the whole of the data array into compressed?

Reply:
Hi @carmenfan , my understanding is that the [filtering_streambuf](https://www.boost.org/doc/libs/1_73_0/libs/iostreams/doc/classes/filtering_streambuf.html) will perform the compression during the iteration (insert into compressed); the array_source makes fullDataArray appear as a [device](https://www.boost.org/doc/libs/1_74_0/libs/iostreams/doc/classes/array.html). As the "device" is unbuffered, I am not sure what would happen if we tried to write the compressed data back into fullDataArray.
(If I understood what you meant correctly of course!)

Reply:
@sebjf Ah ok, I didn't realised i'ts doing a compression whilst inserting. It just looked like we're inserting a massive buffer into a tiny buffer instead of the other way round!